js/apiv2.js

```diff
@@ -139,24 +139,23 @@ const api = {
             })
     },
     async uploadTemplate(program_id, file) {
-        // // TODO
-        //     // Send to backend
-        // console.log("API request to send Templates");
-        //     let valid = Math.ceil(Math.random() * 10); // Mock valid indicators for testing
-        //     let invalid = Math.floor(Math.random() * 2); // Mock invalid indicators for testing
-        //     return await Promise.resolve( {statusText: "OK", data: {valid: valid, invalid: invalid}} )
-        //         .then(response => new Promise( resolve => {
-        //             // Mock varied delayed response from the backend to see variation of the loading spinner. Will be removed once it is actually connected to the backend
-        //             let timeOptions = [500, 900, 1000, 2000, 3000]
-        //             let delay = timeOptions[Math.floor(Math.random() * 5)]
-        //             setTimeout(() => {
-        //                 resolve( response.data )
-        //             }, delay);
-        //         }))
-        //         .catch((error) => {
-        //             this.logFailure(error)
-        //             return {error};
-        //         })
+        // USED FOR TESTING
+        // let valid = Math.ceil(Math.random() * 10); // Mock valid indicators for testing
+        // let invalid = Math.floor(Math.random() * 2); // Mock invalid indicators for testing
+        // return await Promise.resolve( { status: 400,  data: {error_codes: [100, 101], valid: valid, invalid: invalid}} ) // Change status code to test success or failure scenarios
+        //     .then(response => new Promise( resolve => {
+        //         // Mock varied delayed response from the backend to see variation of the loading spinner. Will be removed once it is actually connected to the backend
+        //         let timeOptions = [500, 900, 1000, 2000, 3000]
+        //         let delay = timeOptions[Math.floor(Math.random() * 5)]
+        //         setTimeout(() => {
+        //             resolve( response )
+        //         }, delay);
+        //     }))
+        //     .catch((error) => {
+        //         this.logFailure(error)
+        //         return {error};
+        //     })
+
```

Comment:
Would it make sense to put this in some unit tests? Or maybe you've already done that.

Reply:
Yup, I have another branch that I'm working on testing and will include tests for the different response scenarios.